plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/token/impl/CachedJwtTokenManager.java

```diff
@@ -0,0 +1,245 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.plugin.auth.impl.token.impl;
+
+import com.alibaba.nacos.plugin.auth.exception.AccessException;
+import com.alibaba.nacos.plugin.auth.impl.token.TokenManager;
+import com.alibaba.nacos.plugin.auth.impl.users.NacosUser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Cached JWT token manager.
+ *
+ * @author majorhe
+ */
+@Component
+public class CachedJwtTokenManager implements TokenManager {
+    
+    //key: token string, value: token entity
+    private volatile Map<String, TokenEntity> tokenMap = new ConcurrentHashMap<>(1024);
+    
+    //key: username, value: token entity. cache token created by self.
+    private volatile Map<String, TokenEntity> userMap = new ConcurrentHashMap<>(128);
+    
+    @Autowired
+    private JwtTokenManager jwtTokenManager;
+    
+    @Scheduled(initialDelay = 30000, fixedDelay = 60000)
+    private void cleanExpiredToken() {
+        List<String> tokens = new ArrayList<>();
+        tokenMap.forEach((k, v) -> {
+            if (v.getExpiredTimeMills() < System.currentTimeMillis()) {
+                tokens.add(k);
+            }
+        });
+        tokens.forEach(e -> tokenMap.remove(e));
+        List<String> users = new ArrayList<>();
+        userMap.forEach((k, v) -> {
+            if (v.getExpiredTimeMills() < System.currentTimeMillis()) {
+                users.add(k);
+            }
+        });
+        users.forEach(e -> userMap.remove(e));
+    }
+    
+    /**
+     * Create token.
+     *
+     * @param authentication auth info
+     * @return token
+     */
+    @Override
+    public String createToken(Authentication authentication) throws AccessException {
+        return createToken(authentication.getName());
+    }
+    
+    /**
+     * Create token.
+     *
+     * @param username auth info
+     * @return token
+     */
+    public String createToken(String username) throws AccessException {
+        if (userMap.containsKey(username)) {
+            String token = userMap.get(username).getToken();
+            long expiredTime = userMap.get(username).getExpiredTimeMills();
+            if (!needRefresh(expiredTime)) {
+                return token;
+            }
+        }
+        String token = jwtTokenManager.createToken(username);
+        NacosUser user = jwtTokenManager.parseToken(token);
+        long expiredTime = System.currentTimeMillis() + TimeUnit.SECONDS.toMillis(getTokenValidityInSeconds());
+        Authentication authentication = jwtTokenManager.getAuthentication(token);
```

Comment:
I don't think we need to save `Authentication` always.
`Authentication` will only be used if the authentication plugin is enabled and the plugin name is not `nacos` or `ldap`

Reply:
I really didn't understand the meaning and purpose of Authentication, I just saw that NacosAuthManager called this method.
If it is determined to be useless, can you submit another PR after this PR merge?